pyo3-macros-backend/src/introspection.rs

```diff
@@ -179,20 +182,44 @@ fn argument_introspection_data<'a>(
             IntrospectionNode::String(desc.default_value().into()),
         );
     }
+    if desc.from_py_with.is_none() {
+        // If from_py_with is set we don't know anything on the input type
+        if let Some(ty) = desc.option_wrapped_type {
+            // Special case to properly generate a `T | None` annotation
+            let ty = remove_bound_lifetimes(ty.clone());
+            params.insert(
+                "annotation",
+                IntrospectionNode::InputType {
+                    rust_type: ty,
+                    nullable: true,
+                },
+            );
+        } else {
+            let ty = remove_bound_lifetimes(desc.ty.clone());
+            params.insert(
+                "annotation",
+                IntrospectionNode::InputType {
+                    rust_type: ty,
+                    nullable: false,
```

Comment:
Where's the use case for `nullable`? In entries for `Option` below, it looked like they already encoded the nullability.

Reply:
This is a workaround because of the limitation [described above](https://github.com/PyO3/pyo3/pull/5089#discussion_r2086319331) that prevents to have a proper `T | None` `PYTHON_TYPE` given in the  `impl PyFunctionArgument<'a, 'py, true> for Option<T>`. If I remove this code we would end up with `Any | None` and not `T | None` like here.